# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/EditorCommands.h
# Commit: dd9794f7d400
# Full Hash: dd9794f7d400231a0a06a41ff56e671f05fb825e
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2019-06-10 21:47:35
# Regressor Bug: 1529884
# File Overlap Count: 1
# Description:
#   Bug 1529884 - part 1: Make Document use hashtable to map HTML command and internal command data r=smaug
#   
#   Currently, `Document` converts HTML command (e.g., used by `execCommand()`) to
#   internal XUL command with array in the global space.  However, it requires scan
#   of the array for every command access.
# ==============================================================================

diff -r 66cd074a9182 -r dd9794f7d400 editor/libeditor/EditorCommands.h
--- a/editor/libeditor/EditorCommands.h	Mon Jun 10 10:23:26 2019 +0000
+++ b/editor/libeditor/EditorCommands.h	Mon Jun 10 10:23:59 2019 +0000
@@ -301,6 +301,19 @@
       case Command::FormatIncreaseZIndex:
         return EditorCommandParamType::None;
 
+      // nsClipboardGetContentsCommand
+      // XXX nsClipboardGetContentsCommand is implemented by
+      //     nsGlobalWindowCommands.cpp but cmd_getContents command is not
+      //     used internally, but it's accessible from JS with
+      //     queryCommandValue(), etc.  So, this class is out of scope of
+      //     editor module for now but we should return None for making
+      //     Document code simpler.  We should reimplement the command class
+      //     in editor later for making Document's related methods possible
+      //     to access directly.  Anyway, it does not support `DoCommand()`
+      //     nor `DoCommandParams()` so that let's return `None` here.
+      case Command::GetHTML:
+        return EditorCommandParamType::None;
+
       default:
         MOZ_ASSERT_UNREACHABLE("Unknown Command");
         return EditorCommandParamType::None;
@@ -474,7 +487,7 @@
 
 #define NS_INLINE_DECL_EDITOR_COMMAND_MAKE_SINGLETON(_cmd) \
  public:                                                   \
-  static _cmd* GetInstance() {                             \
+  static EditorCommand* GetInstance() {                    \
     if (!sInstance) {                                      \
       sInstance = new _cmd();                              \
     }                                                      \